cmd/root.go

```diff
@@ -117,15 +119,19 @@ func initViperProfile() {
 		l.Debug().Msgf("Using configuration profile: %s", configFileActiveProfile)
 	}
 
-	// Configure the profile viper instance
-	if err := profiles.GetMainConfig().ChangeActiveProfile(configFileActiveProfile); err != nil {
+	// Configure the profile koanf instance
+	if err := profiles.GetKoanfConfig().ChangeActiveProfile(configFileActiveProfile); err != nil {
 		output.UserFatal(fmt.Sprintf("Failed to set active profile: %v", err), nil)
 	}
 
 	// Validate the configuration
 	if err := profiles.Validate(); err != nil {
 		output.UserFatal(fmt.Sprintf("%v", err), nil)
 	}
+
+	if err := profiles.GetKoanfConfig().WriteFile(); err != nil {
```

Comment:
Why is this WriteFile added here? `checkCfgFileLocation` and `initKoanf` should have handled the file creation/validation.